### PR TITLE
Feature/move to namespace

### PR DESCRIPTION
**Pull request by ryzngard**
**Implement MoveToNamespace feature**

Move to namespace allows a user to move all symbols in a context
to a different namespace, whether it's a new one or existing. This is
different from renaming a namespace as not all symbols in the namespace
are moved, only ones within the current document.

It relies on the IChangeNamespaceService implementation to do the heavy
lifting. Since the current implementation only supports C#, this new
feature will as well.

After this is checked in, when a user has the cursor inside a namespace
declaration that has no nested namespace declarations within it, a code
fix will be suggested to pop up a dialog and help the user choose a new
namespace.

#32314 

image

image

### CODE REVIEW

File: src/VisualStudio/Core/Def/Implementation/MoveToNamespace/VisualStudioMoveToNamespaceOptionsService.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            _threadingContext = threadingContext;
+        }
+
+        public async Task<MoveToNamespaceOptionsResult> GetChangeNamespaceOptionsAsync(
+            ISyntaxFactsService syntaxFactsService,
+            INotificationService notificationService,
+            string defaultNamespace,
+            CancellationToken cancellationToken)
+        {
+            await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync();
```
Comment (sharwell) [inline src/VisualStudio/Core/Def/Implementation/MoveToNamespace/VisualStudioMoveToNamespaceOptionsService.cs]: ❗️ Missing `cancellationToken` use:

```suggestion
            await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
            cancellationToken.ThrowIfCancellationRequested();
```

File: src/VisualStudio/Core/Def/Implementation/MoveToNamespace/MoveToNamespaceDialog.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    /// <summary>
+    /// Interaction logic for MoveToNamespaceDialog.xaml
+    /// </summary>
+    internal partial class MoveToNamespaceDialog : DialogWindow
+    {
+        /// <summary>
+        /// For test purposes only. The integration tests need to know when the dialog is up and
+        /// ready for automation.
+        /// </summary>
+        internal static event Action TEST_DialogLoaded;
```
Comment (sharwell) [inline src/VisualStudio/Core/Def/Implementation/MoveToNamespace/MoveToNamespaceDialog.xaml.cs]: ❗️ This is not required. We can discuss alternatives separately.

**Comment by CyrusNajmabadi** [comment]
Marked as 'personal' right now.  Can you let me know when this is ready for review?  description/images would be nice as well at that point.  Thanks!

**Comment by ryzngard** [comment]
@CyrusNajmabadi @sharwell  this is ready for review 🎉

**Comment by ryzngard** [comment]
I'll reopen as a separate review post UX approval

**Pull request closed without merging**